Utility/NuGetPathUtility.cs

```diff
@@ -124,7 +157,7 @@ internal static Uri ValidateUri(Uri query, NuGetRequest request)
                 // If the query feed exists, it's a non-local repo
                 // Check the query feed to make sure it's available
                 // Optionally we could change this to check the packages feed for availability
-                if (repo.ResourceProvider.QueryFeed == null || !repo.ResourceProvider.QueryFeed.IsAvailable(new RequestWrapper(request)))
+                if (repo.ResourceProvider.QueryFeed == null || !repo.ResourceProvider.QueryFeed.IsAvailable(new RequestWrapper(request, credentials.UserName, credentials.SecurePassword)))
```

Comment:
>credentials.UserName, credentials.SecurePassword [](start = 136, length = 48)

can we just pass in credentials object so we do not have to worry about whether credentials is null or not?